subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/changes/DefaultTaskExecutionModeResolverTest.groovy

```diff
@@ -47,29 +47,16 @@ class DefaultTaskExecutionModeResolverTest extends Specification {
         _ * outputs.getUpToDateSpec() >> upToDateSpec
     }
 
-    def "no actions with no outputs"() {
+    def "no outputs"() {
         when:
         TaskExecutionMode state = repository.getExecutionMode(task, taskProperties)
 
         then:
-        state == NO_OUTPUTS_WITHOUT_ACTIONS
-        1 * upToDateSpec.isEmpty() >> true
+        state == NO_OUTPUTS
         1 * taskProperties.hasDeclaredOutputs() >> false
-        1 * task.hasTaskActions() >> false
-    }
-
-    def "no actions with outputs"() {
-        def nonIncrementalTaskAction = Mock(InputChangesAwareTaskAction)
-
-        when:
-        TaskExecutionMode state = repository.getExecutionMode(task, taskProperties)
-
-        then:
-        state == NO_OUTPUTS_WITH_ACTIONS
         1 * upToDateSpec.isEmpty() >> true
-        1 * taskProperties.hasDeclaredOutputs() >> false
-        1 * task.hasTaskActions() >> true
-        1 * task.getTaskActions() >> [nonIncrementalTaskAction]
+        _ * task.getTaskActions() >> []
```

Comment:
At this point, the task always has actions, so it is strange to return an empty list here.